Sources/Networking/Core/Upload/UploadTask.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  UploadTask.swift
+//  
+//
+//  Created by Tony Ngo on 12.06.2023.
+//
+
+import Combine
+import Foundation
+
+/// Represents and manages an upload task and provides its state.
+public struct UploadTask {
+    public typealias ID = String
+
+    /// The session task this object represents.
+    var task: URLSessionUploadTask
+
+    /// The request associated with this task.
+    let endpointRequest: EndpointRequest
+
+    /// The uploadable data associated with this task.
+    let uploadable: Uploadable
+
+    /// Use this publisher to emit a new state of the task.
+    let statePublisher: CurrentValueSubject<State, Never>
```

Comment:
just throwing idea here... 
this property is publicly used just to pass new value from URLSessionTaskDelegate... in all other cases sending values is extension below. What if we would change this accessory level to private and make public method to pass value into it? similarly we have other methods in extension?

Reply:
The publisher is accessible only within the library, not publicly, whilst the methods in the extension below are. 

> What if we would change this accessory level to private and make public method to pass value into it? similarly we have other methods in extension?

I don't think allowing the client to send new states is a good idea. This publisher should stay an internal implementation detail IMO. Wdyt?